### DISCUSSION TITLE
Is there some builtin way to detect the currently executing shell?

### DISCUSSION THREAD

**Discussion by helpermethod** [Homebrew/discussions#2851] [Tap maintenance and Homebrew development]
It seems that Homebrew can only detect the `preferred` shell, which is not necessarily the shell currently in use.
Is there some way to detect the currently executing shell?

**Comment by gromgit** [comment]
If by "currently executing shell" you mean "the shell that spawned `brew`", then `brew shellenv` already does so, albeit in a Bash script: https://github.com/Homebrew/brew/blob/master/Library/Homebrew/cmd/shellenv.sh

That said, what problem are you trying to solve, that "`brew` detecting the shell executable that spawned it" is the desired answer?
